### PR TITLE
build: Convert build-tools dir to a tsc project

### PR DESCRIPTION
**Pull request by scinos**
#### Background

We have a TypeScript project in `./client` (i.e. it has a `tsconfig.json` file). This project declares `rootDir: '.'` but includes files (`../build-tools/**/*`) from outside the root dir. This is not allowed in TypeScript.

image

#### Changes proposed in this Pull Request

* Make `./build-tools` a separate TypeScript project by creating its own `tsconfig.json`
* Link it to `./client/tsconfig.json` by using `references`
* Refactor `./build-tools/webpack/sections-loader.js` so it receives the config from the loader config, instead of importing a file outside `./build-tools` (namely `../../client/server/config`)

#### Testing instructions

* Run `yarn tsc --project build-tools`, it should compile without errors and generate types in `./build-tools/dist/`
* Start the project with `yarn start`, all sections should be loaded.
* Start the project with `SECTION_LIMIT=me yarn start`. The console output should say it is being built with limited sections, and only http://calypso.localhost:3000/me should work.

### CODE REVIEW

File: packages/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
 	"files": [],
+	"compilerOptions": {
```
Comment (scinos) [inline packages/tsconfig.json:3]: Drive-by fix, `tsc` was complaining about this missing option

**Review by noahtallen — APPROVED** [review]
Tests well for me!
